packages/wallet/src/redux/channelState/ledger-channel/opening/reducer.ts

```diff
@@ -0,0 +1,190 @@
+  }
+};
+
+const waitForInitialPreFundSetupReducer = (
+  state: channelStates.WaitForInitialPreFundSetup,
+  action: internalActions.OpenLedgerChannel | actions.WalletAction,
+): StateWithSideEffects<
+  channelStates.WaitForInitialPreFundSetup | channelStates.WaitForFundingApproval
+> => {
+  switch (action.type) {
+    case actions.COMMITMENT_RECEIVED:
+      const { commitment, signature } = action;
+      const ourIndex = commitment.channel.participants.indexOf(state.address);
+      validatePreFundSetupCommitment(
+        action.commitment,
+        action.signature,
+        ourIndex,
+        commitment.channel.participants,
+      );
+      const {
+        preFundSetupCommitment,
+        commitmentSignature,
+        sendCommitmentAction,
+      } = composePreFundCommitment(
+        commitment.channel,
+        commitment.allocation,
+        commitment.destination,
+        ourIndex,
+        state.privateKey,
+      );
+      return {
+        state: channelStates.waitForFundingApproval({
+          ...state,
+          channelId: channelID(commitment.channel),
+          libraryAddress: commitment.channel.channelType,
+          channelNonce: commitment.channel.nonce,
+          ourIndex,
+          funded: false,
+          participants: commitment.channel.participants as [string, string],
+          turnNum: commitment.turnNum + 1,
+          penultimateCommitment: { commitment, signature },
+          lastCommitment: {
+            commitment: preFundSetupCommitment,
+            signature: commitmentSignature,
+          },
+        }),
+        outboxState: {
+          displayOutbox: showWallet(),
+          messageOutbox: sendCommitmentAction,
+        },
+      };
+  }
+  return { state };
+};
+const waitForPreFundSetupReducer = (
+  state: channelStates.WaitForPreFundSetup,
+  action: internalActions.OpenLedgerChannel | actions.WalletAction,
+): StateWithSideEffects<
+  channelStates.WaitForPreFundSetup | channelStates.WaitForFundingApproval
+> => {
+  switch (action.type) {
+    case actions.COMMITMENT_RECEIVED:
+      validatePreFundSetupCommitment(
+        action.commitment,
+        action.signature,
+        state.ourIndex,
+        state.participants,
+      );
+
+      return {
+        state: channelStates.waitForFundingApproval({
+          ...state,
+          penultimateCommitment: state.lastCommitment,
+          lastCommitment: { commitment: action.commitment, signature: action.signature },
+        }),
+        outboxState: { displayOutbox: showWallet() },
+      };
+  }
+  return { state };
+};
+const sendInitialPreFundSetupReducer = (
+  state: channelStates.SendInitialPreFundSetup,
+  action: internalActions.OpenLedgerChannel | actions.WalletAction,
+): StateWithSideEffects<
+  channelStates.WaitForPreFundSetup | channelStates.SendInitialPreFundSetup
+> => {
+  switch (action.type) {
+    case internalActions.OPEN_LEDGER_CHANNEL:
+      const { channelNonce, ourIndex, participants } = state;
```

Comment:
Why doesn't the state just have a `Channel` property?

Reply:
I'm using the existing `FirstCommitmentReceived` state that is shared by all `channel` reducers.